client/finality-grandpa/src/communication/mod.rs

```diff
@@ -667,19 +693,12 @@ impl<Block: BlockT> Sink<Message<Block>> for OutgoingMessages<Block>
 		}
 
 		// when locals exist, sign messages on import
-		if let Some(ref public) = self.local_id {
-			let keystore = match &self.keystore {
-				Some(keystore) => keystore.clone(),
-				None => {
-					return Err(Error::Signing("Cannot sign without a keystore".to_string()))
```

Comment:
So, if we can not sign, we should not return an error anymore?

Reply:
This branch no longer needs to be handled since it cannot exist. We either have a local id and a keystore or we don't have anything (previously we needed this branch since we had them in two separate `Option`s even though if one was defined the other was always as well).